src/main/java/org/springframework/data/elasticsearch/core/mapping/SimpleElasticsearchPersistentEntity.java

```diff
@@ -102,26 +94,13 @@ public SimpleElasticsearchPersistentEntity(TypeInformation<T> typeInformation) {
 		}
 	}
 
-	@Override
-	public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
-		context.addPropertyAccessor(new BeanFactoryAccessor());
-		context.setBeanResolver(new BeanFactoryResolver(applicationContext));
-		context.setRootObject(applicationContext);
-	}
-
 	private String getIndexName() {
-
-		if (indexName != null) {
-			Expression expression = parser.parseExpression(indexName, ParserContext.TEMPLATE_EXPRESSION);
-			return expression.getValue(context, String.class);
-		}
-
-		return getTypeInformation().getType().getSimpleName();
+		return indexName != null ? indexName : getTypeInformation().getType().getSimpleName();
 	}
 
 	@Override
 	public IndexCoordinates getIndexCoordinates() {
-		return IndexCoordinates.of(getIndexName());
+		return indexCoordinatesResolver.resolve(getEvaluationContext(null), IndexCoordinates.of(getIndexName()));
```

Comment:
How about simplifying this method back to `IndexCoordinates.of(getIndexName()))` and keeping the evaluation within `getIndexName()`?

Reply:
see above